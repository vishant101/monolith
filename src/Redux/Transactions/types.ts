export interface TransactionState {
  status: string;
  favourites: Array<number>;
  processedTransactions: ProcessedTransactions;
  error: string;
}

export interface Transaction {
  user_id: string;
  timestamp: string;
  currency: string;
  amount: string;
}

export interface Transactions extends Array<Transaction> {}

export interface Balances {
  [currency: string]: string;
}

export interface ProcessedTransactionProps {
  user_id: string;
  lastupdated: string;
  transactions: Transactions;
  balances: Balances;
}

export interface ProcessedTransactions {
  [user_id: string]: ProcessedTransactionProps;
}
