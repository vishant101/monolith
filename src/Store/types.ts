import {TransactionState} from '../Redux/Transactions/types';

export interface RootState {
  transactionState: TransactionState;
}
