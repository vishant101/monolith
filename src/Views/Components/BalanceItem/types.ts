import {Balances} from '../../../Redux/Transactions/types';

export interface Props {
  user_id: string;
  balances: Balances;
  timestamp: string;
  onPress: () => void;
}
