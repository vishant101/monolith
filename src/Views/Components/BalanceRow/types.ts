export interface Props {
  currency: string;
  balance: string;
}
