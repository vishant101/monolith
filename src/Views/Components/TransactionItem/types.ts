export interface Props {
  currency: string;
  amount: string;
  timestamp: string;
}
