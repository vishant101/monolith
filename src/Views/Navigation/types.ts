export type StackParamList = {
  Balances: undefined;
  Transactions: {user_id: string};
};
