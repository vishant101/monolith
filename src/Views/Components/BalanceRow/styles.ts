import styled from 'styled-components/native';

export const Container = styled.View`
  flex-direction: row;
  padding: 4px 0px;
  background-color: white;
`;
