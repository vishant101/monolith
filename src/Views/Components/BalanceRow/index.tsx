import React from 'react';
import {Image} from 'react-native';
import {getCurrencyImgSource} from '../../../Utils';
import {Container, CurrencyText, styles} from './styles';
import {Props} from './types';

const BalanceRow = ({currency, balance}: Props) => (
  <Container>
    <Image
      source={getCurrencyImgSource(currency)}
      style={styles.image}
      resizeMode="contain"
    />
    <CurrencyText>{`${currency}: ${balance}`}</CurrencyText>
  </Container>
);

export default BalanceRow;
