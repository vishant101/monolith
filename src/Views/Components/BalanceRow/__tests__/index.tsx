import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import BalanceRow from '..';
import {Props} from '../types';

describe('BalanceRow Component', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      currency: 'EUR',
      balance: '1000',
    };
  });

  it('should render successfully', () => {
    expect(render(<BalanceRow {...props} />).toJSON()).toBeTruthy();
  });

  it('should match the previous snapshot', () => {
    expect(renderer.create(<BalanceRow {...props} />)).toMatchSnapshot();
  });
});
