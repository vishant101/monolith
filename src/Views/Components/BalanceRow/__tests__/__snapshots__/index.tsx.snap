// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceRow Component should match the previous snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 4,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 4,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/Assets/Img/ic_eur.png",
      }
    }
    style={
      Object {
        "height": 24,
        "resizeMode": "stretch",
        "width": 24,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
        },
      ]
    }
  >
    EUR: 1000
  </Text>
</View>
`;
